src/libcollections/string.rs

```diff
@@ -100,6 +101,31 @@ impl String {
         String { vec: ::slice::SliceExt::to_vec(string.as_bytes()) }
     }
 
+    #[cfg(not(stage0))]
+    /// Creates a new string buffer from the given string.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// let s = String::from_str("hello");
+    /// assert_eq!(s.as_slice(), "hello");
+    /// ```
+    #[inline]
+    #[unstable(feature = "collections",
+               reason = "needs investigation to see if to_string() can match perf")]
+    #[cfg(not(test))]
+    pub fn from_str(string: &str) -> String {
```

Comment:
oh, well, I guess they already exist -- I didn't (and don't) see them in the deletions though, which is weird.